airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/avro/JsonSchemaAvroPreprocessor.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.destination.s3.avro
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import io.airbyte.cdk.integrations.destination.s3.jsonschema.AirbyteJsonSchemaType
+import io.airbyte.cdk.integrations.destination.s3.jsonschema.JsonSchemaIdentityMapper
+import io.airbyte.commons.jackson.MoreMappers
+
+class JsonSchemaAvroPreprocessor : JsonSchemaIdentityMapper() {
+    companion object {
+        val STRING_TYPE: ObjectNode =
+            MoreMappers.initMapper().createObjectNode().put("type", "string")
+    }
+
+    override fun mapObjectWithoutProperties(schema: ObjectNode): JsonNode {
+        return STRING_TYPE
+    }
+
+    override fun mapArrayWithoutItems(schema: ObjectNode): JsonNode {
+        return STRING_TYPE
+    }
+
+    override fun mapArrayWithItem(schema: ObjectNode): JsonNode {
+        val items = schema["items"] as ObjectNode
+        val itemType = AirbyteJsonSchemaType.fromJsonSchema(items)
+
+        // Promote an array of unions (unsupported) to an array of mixed types (supported)
```

Comment:
Yes, I'm talking about `items: oneOf`, which our legacy avro-schema converter decides it can't handle and forces to an array of strings.

Avro supports the concept, but it needs to be translated from json schema into something like `{type: array, items: { "type": [ ... list of types ... ]}`.